client/my-sites/checkout/controller.jsx

```diff
@@ -46,7 +46,15 @@ import { getProductSlugFromContext, isContextJetpackSitelessCheckout } from './u
 const debug = debugFactory( 'calypso:checkout-controller' );
 
 export function checkoutJetpackSiteless( context, next ) {
-	sitelessCheckout( context, next, { sitelessCheckoutType: 'jetpack' } );
+	const connectAfterCheckout = context.query?.connect_after_checkout === 'true';
+	const fromSiteSlug = context.query?.from_site_slug;
+	const adminUrl = context.query?.admin_url;
+	sitelessCheckout( context, next, {
+		sitelessCheckoutType: 'jetpack',
+		connectAfterCheckout,
+		...( fromSiteSlug && { fromSiteSlug } ),
+		...( adminUrl && { adminUrl } ),
```

Comment:
Curious - what's the advantage of using a spread operator with a conditional here? Would it work to just pass `fromSiteSlug` and `adminUrl` directly?

Reply:
> Curious - what's the advantage of using a spread operator with a conditional here? Would it work to just pass `fromSiteSlug` and `adminUrl` directly?

Yes, I guess no advantage really..  My thinking was, if the query args are present in the url, then pass the props, otherwise don't.  I guess the only difference would be that if you were  inspecting `extraProps` at this point, you would only see properties that are defined (with values), there wouldn't be properties present with value 'undefined'.